torch/multiprocessing/queue.py

```diff
@@ -22,7 +22,8 @@ def recv(self):
         return pickle.loads(buf)
 
     def __getattr__(self, name):
-        return getattr(self.conn, name)
+        conn = object.__getattribute__(self, 'conn')
```

Comment:
Does this really help? I'm not convinced that this really fixes the problem. IIRC `__getattribute__` will just call `__getattr__` if it doesn't find it in the `__dict__` so you can still end up with this loop. A better fix would be
```python
if 'conn' in self.__dict__:
    return getattr(self.conn, name)
raise AttributeError('no attribute: ' + name)
```

Reply:
@apaszke, your code is better, my code can work. but the origin code is failed.

testcode.py
```python
import torch
import torch.utils.data as Data
import torch.nn.functional as F
from torch.autograd import Variable

if __name__ == '__main__':
    torch.manual_seed(1)    # reproducible
    
    LR = 0.01
    BATCH_SIZE = 32
    EPOCH = 12
    
    # fake dataset
    x = torch.unsqueeze(torch.linspace(-1, 1, 1000), dim=1)
    y = x.pow(2) + 0.1*torch.normal(torch.zeros(*x.size()))
    
    # put dateset into torch dataset
    torch_dataset = Data.TensorDataset(data_tensor=x, target_tensor=y)
    loader = Data.DataLoader(dataset=torch_dataset, batch_size=BATCH_SIZE, shuffle=True, num_workers=2,)
```